lib/pbench/test/functional/agent/cli/commands/tools/test_clear_tools.py

```diff
@@ -362,54 +371,70 @@ def test_clear_tools_test86(tools_configuration, groups, remotes, tools):
     out, err, exitcode = pytest.helpers.capture(command)
     assert b"" == out
 
-    empty_group_err_msg = "Blank group name specified; skipping."
-    empty_remote_err_msg = "Blank remote host name specified; skipping."
-    empty_tool_err_msg = "Blank tool name specified; skipping."
+    empty_group_err_msg = (
+        "Error: Invalid value for '-g' / '--group' / '--groups': "
+        "Blank group name specified; terminating.\n"
+    )
+    empty_remote_err_msg = (
+        "Error: Invalid value for '-r' / '--remote' / '--remotes': "
+        "Blank remote name specified; terminating.\n"
+    )
+    empty_tool_err_msg = (
+        "Error: Invalid value for '-n' / '--name' / '--names': "
+        "Blank tool name specified; terminating.\n"
+    )
 
     if not groups:
         groups = "default"
 
-    empty_group_seq = 0
-    empty_remote_seq = 0
-    empty_tool_seq = 0
-    for group in groups.split(","):
-        invalid_group_err_msg = f'pbench-clear-tools: invalid --group option "{group}"'
-        if not group:
-            empty_group_seq += 1
-        elif group not in tools_tree.keys():
-            assert invalid_group_err_msg.encode() in err
+    groups_list = groups.split(",")
+    remotes_list = remotes.split(",")
+    tools_list = tools.split(",")
+
+    # Check for the blank group name
+    if [group for group in groups_list if not group]:
+        assert empty_group_err_msg.encode() in err
+    else:
+        # Check for the blank remote host name
+        if remotes and [remote for remote in remotes_list if not remote]:
+            assert empty_remote_err_msg.encode() in err
         else:
-            assert invalid_group_err_msg.encode() not in err
-
-            remotes_list = remotes.split(",")
-            if not remotes:
-                remotes_list = tools_tree[group].keys()
-            for remote in remotes_list:
-                invalid_remote_err_msg = f'No remote host "{remote}.example.com"'
-                if not remote:
-                    empty_remote_seq += 1
-                elif remote not in tools_tree[group]:
-                    assert invalid_remote_err_msg.encode() in err
-                else:
-                    assert invalid_remote_err_msg.encode() not in err
-                    wrong_tools = []
-                    all_tools_removed = f'All tools removed from group "{group}" on host "{remote}.example.com"'
-                    if not tools:
-                        assert all_tools_removed.encode() in err
+            # Check for the blank tool name
+            if tools and [tool for tool in tools_list if not tool]:
+                assert empty_tool_err_msg.encode() in err
+            else:
+                # If none of the options have any blank value specified
+                for group in groups_list:
```

Comment:
I think this would be nicer if you used `elif`'s:
```
    if not all(groups_list):     # Check for blank entries in the group names
        assert empty_group_err_msg.encode() in err
    elif not all(remotes_list):  # Check for blank entries in the remotes
        assert empty_remote_err_msg.encode() in err
    elif not all(tools_list):    # Check for blank entries in the tool names
        assert empty_tool_err_msg.encode() in err
    else:
        # None of the options have any blank value specified
        for group in groups_list:
        ...
```